jfrog-cli/artifactory/commands/buildinfo/addgit.go

```diff
@@ -1,42 +1,312 @@
+	}
+	if ic.TrackerUrl != "" {
+		// Url should end with '/'
+		if !strings.HasSuffix(ic.TrackerUrl, "/") {
+			ic.TrackerUrl += "/"
+		}
+	}
+
+	// Get issues key group index
+	if !vConfig.IsSet(ConfigIssuesPrefix + "keyGroupIndex") {
+		return errorutils.CheckError(errors.New(fmt.Sprintf(MissingConfigurationError, ConfigIssuesPrefix + "keyGroupIndex")))
+	}
+	ic.KeyGroupIndex, err = strconv.Atoi(vConfig.GetString(ConfigIssuesPrefix + "keyGroupIndex"))
+	if err != nil {
+		return errorutils.CheckError(errors.New(fmt.Sprintf(ConfigParseValueError, ConfigIssuesPrefix + "keyGroupIndex", err.Error())))
 	}
 
-	err = utils.SavePartialBuildInfo(buildName, buildNumber, populateFunc)
+	// Get issues summary group index
+	if !vConfig.IsSet(ConfigIssuesPrefix + "summaryGroupIndex") {
+		return errorutils.CheckError(errors.New(fmt.Sprintf(MissingConfigurationError, ConfigIssuesPrefix + "summaryGroupIndex")))
+	}
+	ic.SummaryGroupIndex, err = strconv.Atoi(vConfig.GetString(ConfigIssuesPrefix + "summaryGroupIndex"))
+	if err != nil {
+		return errorutils.CheckError(errors.New(fmt.Sprintf(ConfigParseValueError, ConfigIssuesPrefix + "summaryGroupIndex", err.Error())))
+	}
+
+	// Get aggregation aggregate
+	ic.Aggregate = false
+	if vConfig.IsSet(ConfigIssuesPrefix + "aggregate") {
+		ic.Aggregate, err = strconv.ParseBool(vConfig.GetString(ConfigIssuesPrefix + "aggregate"))
+		if err != nil {
+			return errorutils.CheckError(errors.New(fmt.Sprintf(ConfigParseValueError, ConfigIssuesPrefix + "aggregate", err.Error())))
+		}
+	}
+
+	// Get aggregation status
+	if vConfig.IsSet(ConfigIssuesPrefix + "aggregationStatus") {
+		ic.AggregationStatus = vConfig.GetString(ConfigIssuesPrefix + "aggregationStatus")
+	}
+
+	return nil
+}
+
+func (ic *IssuesConfiguration) setArtifactoryDetailsFromConfigFile(vConfig *viper.Viper) error {
+	// If serverId is empty, get default config.
+	serverId := vConfig.GetString("issues.serverID")
+	artDetails, err := utilsconfig.GetArtifactoryConf(serverId)
 	if err != nil {
 		return err
 	}
-	log.Info("Collected git revision and remote url for", buildName+"/"+buildNumber+".")
+	ic.ArtDetails = artDetails
+	return nil
+}
+
+type BuildAddGitConfiguration struct {
+	BuildName      string
+	BuildNumber    string
+	DotGitPath     string
+	ConfigFilePath string
+	IssuesConfig   *IssuesConfiguration
+}
+
+type IssuesConfiguration struct {
+	ArtDetails        *utilsconfig.ArtifactoryDetails
+	Regexp            string
+	LogLimit          int
+	TrackerUrl        string
+	TrackerName       string
+	KeyGroupIndex     int
+	SummaryGroupIndex int
+	Aggregate         bool
+	AggregationStatus string
+}
+
+type LogCmd struct {
+	gitPath         string
+	logLimit        int
+	lastVcsRevision string
+}
+
+func (logCmd *LogCmd) GetCmd() *exec.Cmd {
+	var cmd []string
+	cmd = append(cmd, "git")
+	cmd = append(cmd, "log")
+	cmd = append(cmd, "--pretty=format:%s")
+	cmd = append(cmd, "-"+strconv.Itoa(logCmd.logLimit))
```

Comment:
The above lines can be replaced with a one-liner which creates the cmd. For example:
**append(cmd, "--init-script", config.initScript)**

Reply:
Changed.